uni/lib/view/schedule/widgets/schedule_slot.dart

```diff
@@ -113,9 +118,18 @@ class SubjectButtonWidget extends StatelessWidget {
         'UCURR_GERAL.FICHA_UC_VIEW?pv_ocorrencia_id=$occurrId';
   }
 
-  Future<void> _launchURL() async {
-    final url = toUcLink(occurrId);
-    await launchUrl(Uri.parse(url));
+  void _launchURL(BuildContext context) {
+    final courseUnits = Provider.of<ProfileProvider>(context, listen: false)
+        .profile
+        .courseUnits;
+    final correspondingCourseUnit =
+        courseUnits.firstWhere((courseUnit) => courseUnit.occurrId == occurrId);
+    Navigator.push(
+      context,
+      MaterialPageRoute<CourseUnitDetailPageView>(
+        builder: (context) => CourseUnitDetailPageView(correspondingCourseUnit),
+      ),
+    );
```

Comment:
What if the course doesn't exist? For example, @ttoino have this situation where he have lectures from courses units he is not enrolled yet. Yes, its weird. But what should be the behavior here?